### PR TITLE
Add tickets controllers logic

### PR DESCRIPTION
**Pull request by Glipchik**
Add viewmodels, mapper and controllers

### CODE REVIEW

**Review by stas1408123 — APPROVED** [review]

File: AirlineTickets/AirlineTickets.API/Mapper/Profiles/ModelViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public class ModelViewModel : Profile
+    {
+        public ModelViewModel()
+        {
+            CreateMap<AirlineTicket, TicketViewModel>().ReverseMap();
+            CreateMap<AirlineTicket, CreateUpdateTicketViewModel>().ReverseMap();
+            CreateMap<AirlineTicketCity, TicketCityViewModel>().ReverseMap();
+            CreateMap<AirlineTicketCity, CreateUpdateTicketCityViewModel>().ReverseMap();
+            CreateMap<City, CityViewModel>().ReverseMap();
+            CreateMap<City, CreateUpdateCityViewModel>().ReverseMap();
```
Comment (moltenessence) [inline AirlineTickets/AirlineTickets.API/Mapper/Profiles/ModelViewModel.cs:19]: Is it really necessary to use ReverseMap() for ALL view models? (Pay attentiton to create..., update... ViewModels)

File: AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/TicketCityViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace AirlineTickets.API.ViewModels.AirlineTicketCity
+{
+    public class TicketCityViewModel
+    {
+        public int AirlineTicketId { get; set; }
+        public int CityId { get; set; }
+        public CityStayingStatus StayingStatus { get; set; }
+
+        public AirlineTicketEntity? AirlineTicket { get; set; }
+        public CityEntity? City { get; set; }
```
Comment (moltenessence) [inline AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/TicketCityViewModel.cs:13]: Why do u use entity in ViewModel?

Comment (Glipchik) [reply AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/TicketCityViewModel.cs]: fixed

File: AirlineTickets/AirlineTickets.API/Mapper/Profiles/ModelViewModel.cs (hunk 1 above)
Comment (Glipchik) [reply AirlineTickets/AirlineTickets.API/Mapper/Profiles/ModelViewModel.cs]: fixed

File: AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/CreateUpdateTicketCityViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using AirlineTickets.Core.Enums;
+
+namespace AirlineTickets.API.ViewModels.AirlineTicketCity
+{
+    public class CreateUpdateTicketCityViewModel
+    {
+        public CityStayingStatus StayingStatus { get; set; }
```
Comment (GodlyCat) [inline AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/CreateUpdateTicketCityViewModel.cs:7]: Add ids for city and ticket

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            _ticketCityService = ticketCityService;
+            _mapper = mapper;
+        }
+
+        [HttpGet()]
+        public async Task<IEnumerable<TicketCityViewModel>> GetAll(CancellationToken cancellationToken) =>
+            _mapper.Map<IEnumerable<TicketCityViewModel>>(await _ticketCityService.GetAll(cancellationToken));
+
+        [HttpGet("{ticketId}&{cityId}")]
```
Comment (GodlyCat) [inline AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs:26]: Ticket{ticketId}\City{cityId}

**Review by GodlyCat — CHANGES_REQUESTED** [review]

File: AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/CreateUpdateTicketCityViewModel.cs (hunk 3 above)
Comment (Glipchik) [reply AirlineTickets/AirlineTickets.API/ViewModels/AirlineTicketCity/CreateUpdateTicketCityViewModel.cs:9]: fixed

**Review by GodlyCat — APPROVED** [review]

**Pull request merged**